proc/test_thread_rand.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-tests
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	unsigned int time;
+	unsigned int id;
+} threads_self_t;
+
+
+typedef struct {
+	unsigned int cthread;
+	unsigned int fin;
+	char stack[10][512];
+	handle_t fin_lock;
+	threads_self_t params[10];
+} threads_common_t;
+
+
+static threads_common_t threads_common = { .cthread = 10, .fin = 0 };
+
+
+static void test_thread(void *arg)
+{
+	threads_self_t *self = (threads_self_t *)arg;
+	usleep(1000 * self->time);
+
+	mutexLock(threads_common.fin_lock);
```

Comment:
I don't see this mutex being created anywhere. Does this even work!?

Reply:
Surprisingly it works. Added proper creation and destroy of mutex with `mutexCreate()` and `resourceDestroy()`.

Reply:
Actually - You don't know as You're not checking the result code of `mutexLock` so You don't know if it succeeded.

Please add checking result code to all native calls (`mutex*`, `begin/endthread`, `resourceDestroy`)